Libs/RaptureCore/src/main/java/rapture/script/reflex/ReflexRaptureScript.java

```diff
@@ -287,39 +288,100 @@ public ScriptResult runProgramExtended(CallingContext context, IActivityInfo act
     private ScriptResult _doRunProgram(CallingContext context, IActivityInfo activity, RaptureScript script, Map<String, Object> params, int timeout,
             ScriptRunInfoCollector collector) throws RecognitionException, ReflexException {
         ScriptResult res = new ScriptResult();
-            ReflexTreeWalker walker = getParserWithStandardContext(context, script.getScript(), params);
-            ProgressDebugger progress = (timeout > 0) ? new TimeoutReflexDebugger(activity, script.getScript(), timeout)
-                    : new ProgressDebugger(activity, script.getScript());
-            // Setup an alternate output handler, and a standard data handler
-            walker.getReflexHandler().setDataHandler(new ReflexDataHelper(context));
-            walker.getReflexHandler().setOutputHandler(new ScriptResultOutputHandler(res));
-            ReflexNode execRes = walker.walk();
-
-            ReflexExecutor.injectSystemIntoScope(walker.currentScope);
-
-            for (Map.Entry<String, Object> val : params.entrySet()) {
-                log.debug("Looking to inject " + val.getKey());
-                ReflexValue v = walker.currentScope.resolve(val.getKey());
-                if (v != null && v.getValue() != ReflexValue.Internal.VOID && v.getValue() != ReflexValue.Internal.NULL) {
-                    log.debug("Injecting " + v.asObject() + " as " + val.getKey());
-                    val.setValue(v.asObject());
-                } else walker.currentScope.assign(val.getKey(), val.getValue() == null ? new ReflexNullValue() : new ReflexValue(val.getValue()));
-
+        ReflexTreeWalker walker = getParserWithStandardContext(context, script.getScript(), params);
+        ProgressDebugger progress = (timeout > 0) ? new TimeoutReflexDebugger(activity, script.getScript(), timeout)
+                : new ProgressDebugger(activity, script.getScript());
+        // Setup an alternate output handler, and a standard data handler
+        walker.getReflexHandler().setDataHandler(new ReflexDataHelper(context));
+        walker.getReflexHandler().setOutputHandler(new ScriptResultOutputHandler(res));
+        ReflexNode execRes = walker.walk();
+
+        ReflexExecutor.injectSystemIntoScope(walker.currentScope);
+
+        MetaScriptInfo scriptInfo = walker.getScriptInfo();
+
```

Comment:
I don't think so but I'll certainly double check. If there's no meta info for a parameter then they should just be treated as strings as before.